src/components/Elements.test.js

```diff
@@ -83,6 +84,40 @@ describe('Elements', () => {
     expect(wrapper.find(TestComponent).prop('elements')).toBe(mockElements);
   });
 
+  it('works with a Promise resolving to a valid Stripe object', () => {
+    const stripePromise = Promise.resolve(stripe);
+
+    const wrapper = mount(
+      <Elements stripe={stripePromise}>
+        <InjectedTestComponent />
+      </Elements>
+    );
+
+    expect(wrapper.find(TestComponent).prop('elements')).toBe(null);
+
+    return Promise.resolve(act(() => stripePromise)).then(() => {
+      wrapper.update();
+      expect(wrapper.find(TestComponent).prop('elements')).toBe(mockElements);
+    });
+  });
+
+  it('works with a Promise resolving to null for SSR safety', () => {
```

Comment:
Wouldn't a transition from `null` to a Promise also be a valid use case? I believe the code handles it, but we should test for it.

Reply:
Valid but a bit weird. I can add this.